packages/components/src/TreeView/TreeViewItem/TreeViewItem.component.js

```diff
@@ -23,6 +23,28 @@ export function getItemIcon(iconName = 'talend-folder', toggled) {
 	return toggled ? iconName : `${iconName}-closed`;
 }
 
+/**
+ * Internal: you should not use it
+ * Instantiate an icon based on the icon config
+ *
+ * @param icon The icon name of the Icon props
+ * @param toggled if the treeview is toggled
+ */
+function TreeViewIcon({ icon, toggled }) {
+	const isObject = typeof icon === 'object';
+	const iconConfig = isObject ?
+		{ ...icon, className: classNames(css['tc-treeview-img'], icon.className) } :
+		{ name: getItemIcon(icon, toggled), className: css['tc-treeview-folder'] };
+
+	return (
+		<Icon {...iconConfig} />
+	);
```

Comment:
Please use a regular `if/else` statement that returns an `Icon` with the proper props.